src/envelope.rs

```diff
@@ -90,84 +88,57 @@ impl<A: Handler<M, Return = R>, M: Send + 'static, R: Send + 'static> MessageEnv
     }
 }
 
-/// An envelope that does not return a result from a message. Constructed  by the `AddressExt::do_send`
-/// method.
-pub struct NonReturningEnvelope<A, M> {
+/// Special-case envelope for handlers which return `()`.
```

Comment:
Is this actually the case? Currently, this is only used for broadcast, and from this I'd expect it to be used in all cases where `Handler::Return = ()`. Also, there's another difference in that this envelope won't let you wait for a message to be handled, but I'm not sure if this is worth mentioning.

Reply:
True, I guess from these docs one could think that this is always used when handlers return `()` which isn't actually the case.